test_iokit.py

```diff
@@ -122,3 +159,206 @@ def test_IOObjectCopySuperclassForClass():
     class_name = "IOService"
     superclass_name = iokit.corefoundation_to_native(iokit.IOObjectCopySuperclassForClass(class_name))
     assert superclass_name == "IORegistryEntry"
+
+
+def test_IOIteratorNext():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IOService".encode(),
+            iokit.kNilOptions,
+            None
+        )
+    assert err == 0
+
+    obj = iokit.IOIteratorNext(iterator)
+    assert obj != 0 # 0 means the iterator handle is invalid
+
+    err = iokit.IOObjectRelease(obj)
+    assert err == 0
+
+
+def test_IOIteratorIsValid():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IOService".encode(),
+            iokit.kNilOptions,
+            None
+        )
+    assert err == 0
+    assert iokit.IOIteratorIsValid(iterator) == True
+
+
+def test_IORegistryEntryFromPath():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/IOResources".encode()
+    )
+    assert device != iokit.NULL
+
+    err = iokit.IOObjectRelease(device)
+    assert err == 0
+
+
+def test_IORegistryEntryGetLocationInPlane():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/PEGP@0".encode()
+    )
+    assert device != iokit.NULL
+
+    err, location = iokit.IORegistryEntryGetLocationInPlane (
+        device,
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert location.replace(b"\x00", b"") == b"0"
+
+
+def test_IORegistryEntryGetPath():
+    device = _get_IOResources()
+
+    err, path = iokit.IORegistryEntryGetPath (
+        device,
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert path.replace(b"\x00", b"") == b"IOService:/IOResources"
```

Comment:
Use `io_name_t_to_str`
Release the device

Reply:
:+1: